drivers/usb_c/tcpc/ucpd_stm32.c

```diff
@@ -1024,45 +1024,25 @@ static int ucpd_transmit_data(const struct device *dev,
 	return 0;
 }
 
-/**
- * @brief Tests if a received Power Delivery message is pending
- *
- * @retval 0 if there is no pending message
- * @retval 1 if there is a pending message
- */
-static int ucpd_is_rx_pending_msg(const struct device *dev, enum pd_packet_type *type)
-{
-	struct tcpc_data *data = dev->data;
-	bool ret;
-
-	ret = (*(uint32_t *)data->ucpd_rx_buffer > 0);
-
-	if (ret & (type != NULL)) {
-		*type = *(uint16_t *)data->ucpd_rx_buffer;
-	}
-
-	return ret;
-}
-
 /**
  * @brief Retrieves the Power Delivery message from the TCPC
  *
- * @retval number of bytes received
- * @retval -EIO on no message to retrieve
- * @retval -EFAULT on buf being NULL
+ * @retval number of bytes received if msg parameter is provided
+ * @retval 0 if there is a message pending and the msg parameter is NULL
+ * @retval -ENODATA if there is no pending message
  */
-static int ucpd_receive_data(const struct device *dev, struct pd_msg *msg)
+static int ucpd_get_rx_pending_msg(const struct device *dev, struct pd_msg *msg)
 {
 	struct tcpc_data *data = dev->data;
 	int ret = 0;
 
-	if (msg == NULL) {
-		return -EFAULT;
+	/* Make sure we have a message to retrieve */
+	if (*(uint32_t *)data->ucpd_rx_buffer == 0) {
+		return -ENODATA;
 	}
 
-	/* Make sure we have a message to retrieve */
-	if (!ucpd_is_rx_pending_msg(dev, NULL)) {
-		return -EIO;
+	if (!msg) {
```

Comment:
Per the misra r.14.4, you need explicitly check against NULL here.

https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html
https://gitlab.com/MISRA/MISRA-C/MISRA-C-2012/Example-Suite/-/blob/master/R_14_04.c

At one point there was  a compliance check for this, but it got disabled.

Reply:
Done